### PR TITLE
[PUBDEV-7029] Add xgboost offset column

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

**Comment by michalkurka** [comment]
@honzasterba this looks good but I don't see any changes in MOJOs

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/predict/XGboostJavaBigScoreChunkPredict.java [hunk 1]
```diff
@@ -24,8 +24,6 @@ public XGboostJavaBigScoreChunkPredict(DataInfo di, XGBoostOutput output, double
 
   @Override
   public double[] score0(Chunk[] chks, double offset, int row_in_chunk, double[] tmp, double[] preds) {
-    if (offset != 0) throw new UnsupportedOperationException("Unsupported: offset != 0");
```
Comment (michalkurka) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/predict/XGboostJavaBigScoreChunkPredict.java]: offset doesn't seem to be considered

File: h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostNativeMojoModel.java [hunk 2]
```diff
@@ -38,19 +39,24 @@ private static Booster makeBooster(byte[] boosterBytes) {
...
-                                double _defaultThreshold, final boolean _sparse) {
-
+  public static double[] score0(
+      double[] doubles, double offset, double[] preds,
+      String _boosterType, int _ntrees, 
+      Booster _booster, int _nums, int _cats,
+      int[] _catOffsets, boolean _useAllFactorLevels,
+      int nclasses, double[] _priorClassDistrib,
+      double _defaultThreshold, final boolean _sparse, 
+      double _mojo_version
```
Comment (michalkurka) [inline h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostNativeMojoModel.java]: can we avoid making it a parameter? for in H2O execution, we don't really need to add the version

you can move the check to the non-static method (line 42) and throw the message there

Comment (honzasterba) [reply h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostNativeMojoModel.java]: 👍 this was lazy of me

Comment (honzasterba) [reply h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostNativeMojoModel.java]: fixed

**Pull request merged**